src/plugins/photo-library.ts

```diff
@@ -0,0 +1,252 @@
+ *         console.log(libraryItem.photoURL);    // Cross-platform access to photo
+ *         console.log(libraryItem.thumbnailURL);// Cross-platform access to thumbnail
+ *         console.log(libraryItem.fileName);
+ *         console.log(libraryItem.width);
+ *         console.log(libraryItem.height);
+ *         console.log(libraryItem.creationDate);
+ *         console.log(libraryItem.latitude);
+ *         console.log(libraryItem.longitude);
+ *         console.log(libraryItem.albumIds);    // array of ids of appropriate AlbumItem, only of includeAlbumsData was used
+ *       });
+ *     },
+ *     error: err => {},
+ *     complete: () => { console.log("could not get photos"); }
+ *   });
+ * })
+ * .catch(err => console.log("permissions weren't granted"));
+ *
+ * ```
+ */
+@Plugin({
+  pluginName: 'PhotoLibrary',
+  plugin: 'cordova-plugin-photo-library',
+  pluginRef: 'cordova.plugins.photoLibrary',
+  repo: 'https://github.com/terikon/cordova-plugin-photo-library',
+  install: 'ionic plugin add cordova-plugin-photo-library --variable PHOTO_LIBRARY_USAGE_DESCRIPTION="To choose photos"'
+})
+export class PhotoLibrary {
+
+  /**
+   * Retrieves library items. Library item contains photo metadata like width and height, as well as photoURL and thumbnailURL.
+   * @param options {GetLibraryOptions} Optional, like thumbnail size and chunks settings.
+   * @return {Observable<LibraryItem[]>} Returns library items. If appropriate option was set, will be returned by chunks.
+   */
+  @CordovaFiniteObservable({
+    callbackOrder: 'reverse',
+    resultFinalPredicate: (result: {isLastChunk: boolean}) => { return result.isLastChunk; },
+    resultTransform: (result: {library: LibraryItem[]}) => { return result.library; },
+  })
+  static getLibrary(options?: GetLibraryOptions): Observable<LibraryItem[]> { return; }
+
+  /**
+   * Asks user permission to access photo library.
+   * @param options {RequestAuthorizationOptions} Optional, like whether only read access needed or read/write.
+   * @return { Promise<void>} Returns a promise that resolves when permissions are granted, and fails when not.
+   */
+  @Cordova({
+    callbackOrder: 'reverse',
+  })
+  static requestAuthorization(options?: RequestAuthorizationOptions): Promise<void> { return; }
+
+  /**
+   * Returns list of photo albums on device.
+   * @return {Promise<AlbumItem[]>} Resolves to list of albums.
+   */
+  @Cordova({
+    callbackOrder: 'reverse',
+  })
+  static getAlbums(): Promise<AlbumItem[]> { return; }
+
+  /**
+   * Provides means to request URL of thumbnail, with specified size or quality.
+   * @param photo {string | LibraryItem} Id of photo, or LibraryItem.
+   * @param options {GetThumbnailOptions} Options, like thumbnail size or quality.
+   * @return {Promise<string>} Resolves to URL of cdvphotolibrary schema.
+   */
+  static getThumbnailURL(photoId: string, options?: GetThumbnailOptions): Promise<string>;
```

Comment:
Overloads need a `@private` tag so they don't confuse the doc generator. Add the function docs above the `@Cordova` decorator.